org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/code/makestatic/ChangeCalculator.java

```diff
@@ -0,0 +1,502 @@
+	 *         ChangeCalculator.
+	 */
+	public TextEditBasedChange[] getChanges() {
+		return fChangeManager.getAllChanges();
+	}
+
+	/**
+	 * Retrieves the flag indicating whether the target method has instance usage.
+	 *
+	 * @return {@code true} if the target method has instance usage, {@code false} otherwise.
+	 */
+	public boolean getTargetMethodhasInstanceUsage() {
+		return fTargetMethodhasInstanceUsage;
+	}
+
+	/**
+	 * Computes the edit for the target method declaration and adds it to the change manager.
+	 *
+	 * @throws JavaModelException is thrown when the underlying compilation units buffer could not
+	 *             be accessed.
+	 */
+	public void computeMethodDeclarationEdit() throws JavaModelException {
+		//Changes can't be applied to directly to AST, edits are saved in fChangeManager
+		TextEdit methodDeclarationEdit= fTargetMethodDeclarationASTRewrite.rewriteAST();
+		addEditToChangeManager(methodDeclarationEdit, fTargetMethod.getCompilationUnit());
+	}
+
+	/**
+	 * This method uses an InstanceUsageRewriter to rewrite the instance usages in the target method
+	 * declaration. The rewritten AST is updated in the fTargetMethodDeclarationASTRewrite. The
+	 * fTargetMethodhasInstanceUsage flag is also updated based on the instance usage rewriting.
+	 */
+	public void rewriteInstanceUsages() {
+		InstanceUsageRewriter instanceUsageRewriter= new InstanceUsageRewriter(fParameterName, fTargetMethodDeclarationASTRewrite, fTargetMethodDeclarationAST, fStatus, fTargetMethodDeclaration);
+		fTargetMethodDeclaration.getBody().accept(instanceUsageRewriter);
+		fTargetMethodhasInstanceUsage= instanceUsageRewriter.fTargetMethodhasInstanceUsage;
+	}
+
+	/**
+	 * This method uses a ModifierRewrite to add the static modifier to the target method
+	 * declaration. The fTargetMethodDeclarationASTRewrite is updated with the modified AST.
+	 */
+	public void addStaticModifierToTargetMethod() {
+		ModifierRewrite modRewrite= ModifierRewrite.create(fTargetMethodDeclarationASTRewrite, fTargetMethodDeclaration);
+		modRewrite.setModifiers(fTargetMethodDeclaration.getModifiers() | Modifier.STATIC, null);
+	}
+
+	private String generateUniqueParameterName() {
+		String className= ((TypeDeclaration) fTargetMethodDeclaration.getParent()).getName().toString();
+		List<SingleVariableDeclaration> parameters= fTargetMethodDeclaration.parameters();
+		String classNameFirstLowerCase= Character.toLowerCase(className.charAt(0)) + className.substring(1); //makes first char lower to match name conventions
+		if (parameters == null || parameters.isEmpty()) {
+			return classNameFirstLowerCase;
+		}
+		boolean duplicateExists= false;
+		String combinedName= classNameFirstLowerCase;
+		int counter= 2;
+		while (true) {
+			for (SingleVariableDeclaration param : parameters) {
+				String paramString= param.getName().getIdentifier();
+				if (!(combinedName.equals(paramString))) {
+					duplicateExists= false;
+				} else {
+					duplicateExists= true;
+					break;
+				}
```

Comment:
```suggestion
				duplicateExists = combinedName.equals(paramString);
				if (duplicateExists) {
					break;
				}
```

Reply:
I could not commit this change directly from here so I made the commit myself. ae34115fda